UnitTests/Bookmarks/Model/LocalBookmarkManagerTests.swift

```diff
@@ -440,9 +790,141 @@ fileprivate extension Bookmark {
+fileprivate extension BaseBookmarkEntity {
+
+    func matchesBookmark(withTitle title: String, url: URL, isFavorite: Bool, parent: String? = nil) -> Bool {
+        guard let bookmark = self as? Bookmark else { return false }
+        let lhsIsInRoot = bookmark.parentFolderUUID == nil || bookmark.parentFolderUUID == "bookmarks_root" || bookmark.parentFolderUUID == PseudoFolder.bookmarks.id
+        let rhsIsInRoot = parent == nil || parent == "bookmarks_root" || parent == PseudoFolder.bookmarks.id
+        return bookmark.title == title && bookmark.url == url.absoluteString && bookmark.isFavorite == isFavorite && lhsIsInRoot == rhsIsInRoot
+    }
+
+    func matchesBookmark(withTitle title: String, url: URL, isFavorite: Bool, parent: BookmarkFolder? = nil) -> Bool {
+        matchesBookmark(withTitle: title, url: url, isFavorite: isFavorite, parent: parent?.id)
+    }
+
+    func matches(_ entity: BaseBookmarkEntity?) -> Bool {
+        switch entity {
+        case let bookmark as Bookmark:
+            return matchesBookmark(withTitle: bookmark.title, url: URL(string: bookmark.url)!, isFavorite: bookmark.isFavorite, parent: bookmark.parentFolderUUID)
+        case let folder as BookmarkFolder:
+            return matchesFolder(withTitle: folder.title, parent: folder.parentFolderUUID)
+        case .none:
+            return false
+        case .some(let entity):
+            fatalError("Unexpected entity type \(entity)")
+        }
+    }
+
+    func matchesFolder(withTitle title: String, parent: String? = nil) -> Bool {
+        guard self.isFolder else { return false }
+        return self.title == title && self.parentFolderUUID ?? "bookmarks_root" == parent ?? "bookmarks_root"
+    }
+
+    func matchesFolder(withTitle title: String, parent: BookmarkFolder? = nil) -> Bool {
+        matchesFolder(withTitle: title, parent: parent?.id)
+    }
+
+    func matches(_ folder: BookmarkFolder) -> Bool {
+        matchesFolder(withTitle: folder.title, parent: folder.parentFolderUUID)
+    }
 
 }
 
+protocol BookmarksBuilderItem {
+    func build(withParentId parentId: String?) -> BaseBookmarkEntity
+}
+extension BookmarksBuilderItem {
+    func build() -> BaseBookmarkEntity {
+        build(withParentId: nil)
+    }
+}
+extension BaseBookmarkEntity: BookmarksBuilderItem {
+    func build(withParentId parentId: String?) -> BaseBookmarkEntity {
+        switch self {
+        case let bookmark as Bookmark:
+            Bookmark(id: bookmark.id, url: bookmark.url, title: bookmark.title, isFavorite: bookmark.isFavorite, parentFolderUUID: bookmark.parentFolderUUID, faviconManagement: bookmark.faviconManagement)
+        case let folder as BookmarkFolder:
+            BookmarkFolder(id: folder.id, title: folder.title, parentFolderUUID: folder.parentFolderUUID, children: folder.children)
+        default:
+            fatalError("Unexpected entity type \(self)")
+        }
+    }
+}
+private typealias BookmarksBuilder = ArrayBuilder<BookmarksBuilderItem>
```

Comment:
I realize that this builder works on `BaseBookmarkEntity` type and I appreciate you adding it, but just wanted to remind and check if you remembered that we have `BookmarkTreeBuilder` in BSK that works on `BookmarkEntity` (managed object class) and could have perhaps be reused here. I understand if it couldn't 👍